src/analysis/retail/evoker/preservation/normalizers/CastLinkNormalizer.ts

```diff
@@ -584,6 +584,15 @@ const EVENT_LINKS: EventLink[] = [
       return c.hasTalent(TALENTS_EVOKER.SPARK_OF_INSIGHT_TALENT);
     },
   },
+  {
```

Comment:
Might be helpful/easier to event link the tier essence burst procs to the apply and remove buff events of Essence Rush, ID: 409899  and then look for that instead of having to do negative logic checks in your isEBFromTier30 function 
![image](https://user-images.githubusercontent.com/26779541/232226138-9ea65ee3-6248-419a-91dc-d26fa5a4fab6.png)

Reply:
I think this is actually trickier because theoretically someone could cast living flame inside essence rush and proc essence burst but it won't be from the tier set.

Reply:
Oh just saw this, I meant specifically linking to the apply and remove buff events since they occur on the exact time stamp of the procs, not just checking if the buff is active. This works as well so doesn't matter